ili9341 documentation

Valcob opened the pull request:
## Description:


**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#<https://github.com/esphome/esphome/pull/1233>

## Checklist:

  - [x] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [x] Link added in `/index.rst` when creating new documents for new components or cookbook.

Valcob committed:
ili9341 documentation

Valcob committed:
fixed file name

Valcob committed:
change the name one more time

Valcob committed:
more naming fixes

glmnet commented on components/display/ili9341.rst:
```diff
@@ -0,0 +1,123 @@
+    display:
+        ...
+        lambda: |-
+          it.rectangle(0,  0, it.get_width(), it.get_height(), id(my_red);
```
is it missing a `)` at the end?

Valcob replied:
fixed, thnks!

Valcob committed:
add missing parenthesis

glmnet commented on components/display/ili9341.rst:
```diff
@@ -0,0 +1,123 @@
+Configuration variables
+***********************
+
+- **led_pin** (:ref:`Pin Schema <config-pin_schema>`): The display's backlight pin.
```
mark this pin optional

Valcob replied:
Led pin cannot be optional if this pin is not connected to a + rail the screen will be black all the time also it can be connected to a pwm then we can even dim the brightness of the screen

glmnet replied:
I have this pin connected to +VCC via a R 3.3K and it works ok, this saves a pin on the ESP, may be this can be stated on the docs.

glmnet commented on components/display/ili9341.rst:
```diff
@@ -0,0 +1,123 @@
+***********************
+
+- **led_pin** (:ref:`Pin Schema <config-pin_schema>`): The display's backlight pin.
+- **cs_pin** (:ref:`Pin Schema <config-pin_schema>`): The CS pin.
+- **dc_pin** (:ref:`Pin Schema <config-pin_schema>`): The DC pin.
```
those mark required, please sort them by required then optional
also add ``model`` required

Valcob replied:
Hey @glmnet what do you mean by add a model?

glmnet replied:
there is a model property!, marked required see here https://github.com/esphome/esphome/blob/d20caa9d60e91b3044467f16eedfef9ab89d1299/esphome/components/ili9341/display.py#L29

glmnet commented on components/display/ili9341.rst:
```diff
@@ -0,0 +1,123 @@
+- **led_pin** (:ref:`Pin Schema <config-pin_schema>`): The display's backlight pin.
+- **cs_pin** (:ref:`Pin Schema <config-pin_schema>`): The CS pin.
+- **dc_pin** (:ref:`Pin Schema <config-pin_schema>`): The DC pin.
+- **reset_pin** (:ref:`Pin Schema <config-pin_schema>`): The RESET pin.
```
mark this optional, as it is in the schema, also I was testing this and found out that the thing does not work if the pin is not connected (do you know why?)

Valcob replied:
Hi there, what pin are you talking about? reset_pin?

glmnet replied:
sorry, I messed up there, anyway, all pins should be marked either optional or required following the schema definition rules.

glmnet replied:
also, the `reset_pin` it is marked optional on the schema, how is it supposed to work if it is not connected?

glmnet commented on components/display/ili9341.rst:
```diff
@@ -0,0 +1,123 @@
+--------
+
+- :doc:`index`
+- :apiref:`st7789v_base/st7789v_base.h`
```
this is wrong

glmnet commented on components/display/ili9341.rst:
```diff
@@ -0,0 +1,123 @@
+      - file: "image.jpg"
+        id: my_image
+        resize: 200x200
+        type: RGB565
```
```suggestion
        type: RGB24
```

Valcob committed:
Update components/display/ili9341.rst

Co-authored-by: Guillermo Ruffino <glm.net@gmail.com>

Valcob committed:
Update ili9341.rst

fixed: api ref
fixed: led_pin description
marked: all pins as required for a proper functioning

Valcob committed:
Update ili9341.rst

glmnet commented on components/display/ili9341.rst:
```diff
@@ -43,10 +43,10 @@ too (with some SPI flash changes).
-- **cs_pin** (:ref:`Pin Schema <config-pin_schema>`): The CS pin.
-- **dc_pin** (:ref:`Pin Schema <config-pin_schema>`): The DC pin.
-- **reset_pin** (:ref:`Pin Schema <config-pin_schema>`): The RESET pin.
+- **led_pin** (**Required**:ref:`Pin Schema <config-pin_schema>`): The display's backlight pin. It may be connected through a 3.3K resistor to +3V rail this way we can save one pin on ESP
```
this is optional in code, optional / required must match code see https://github.com/esphome/esphome/blob/d20caa9d60e91b3044467f16eedfef9ab89d1299/esphome/components/ili9341/display.py#L32

glmnet reviewed (changes requested):
Please review code and docs that properties are all listed and they requireness, value type, default value etc is all documented property.
Is something does not match you can send another PR to modify code 

Thanks

Valcob committed:
Merge branch 'next' into next

glmnet commented:
Closing as already merged in #870
